### ISSUE TITLE
Allow for removing references in S2 modal after close & re-open

### ISSUE THREAD

**Issue by jvoytek** [refstudio/refstudio#561]
> On a higher level, closing and re-opening the modal resets the state, so if you add a reference, close the modal, open the modal and then search for the same reference again, you are able to add it a second time and you don't have access to the remove button. Maybe that's ok, but I wanted to point it out.

We would need to save the paperId on our end to have any hope of checking for existing references and even then I would be wary of S2's ability to always return the same paperId (I just don't know if or how often those ids change). I think the risk of not being able to add references would be worse than the ability to add multiples of the same reference, but maybe worth a follow up issue...

_Originally posted by @jvoytek in https://github.com/refstudio/refstudio/issues/557#issuecomment-1728055601_

State is reset when the modal closes. This is a larger issue than can be handled in current release.

**Comment by cguedes** [comment]
Agree. I think we need to save the source (type and id) for an ingested reference so that we can know how to dedup for scenarios like this one.
